src/contracts/strategies/StrategyFactory.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import "@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol";
+import "@openzeppelin-upgrades/contracts/access/OwnableUpgradeable.sol";
+import "./StrategyFactoryStorage.sol";
+import "./StrategyBase.sol";
+import "../permissions/Pausable.sol";
+
+/**
+ * @title Factory contract for deploying BeaconProxies of a Strategy contract implementation for arbitrary ERC20 tokens
+ *        and automatically adding them to the StrategyWhitelist in EigenLayer.
+ * @author Layr Labs, Inc.
+ * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
+ * @dev This may not be compatible with non-standard ERC20 tokens. Caution is warranted.
+ */
+contract StrategyFactory is StrategyFactoryStorage, OwnableUpgradeable, Pausable {
+
+    uint8 internal constant PAUSED_NEW_STRATEGIES = 0;
+
+    /// @notice EigenLayer's StrategyManager contract
+    IStrategyManager public immutable strategyManager;
+
+    /// @notice Since this contract is designed to be initializable, the constructor simply sets the immutable variables.
+    constructor(IStrategyManager _strategyManager) {
+        strategyManager = _strategyManager;
+        _disableInitializers();
+    }
+
+    function initialize(
+        address _initialOwner,
+        IPauserRegistry _pauserRegistry,
+        uint256 _initialPausedStatus,
+        IBeacon _strategyBeacon
+    )
+        public virtual initializer
+    {
+        _transferOwnership(_initialOwner);
+        _initializePauser(_pauserRegistry, _initialPausedStatus);
+        _setStrategyBeacon(_strategyBeacon);
+    }
+
+    /**
+     * @notice Deploy a new strategyBeacon contract for the ERC20 token.
+     * @dev A strategy contract must not yet exist for the token.
+     * $dev Immense caution is warranted for non-standard ERC20 tokens, particularly "reentrant" tokens
+     * like those that conform to ERC777.
+     */
+    function deployNewStrategy(IERC20 token) external onlyWhenNotPaused(PAUSED_NEW_STRATEGIES) returns (IStrategy newStrategy) {
+        require(tokenStrategy[token] == IStrategy(address(0)),
+            "StrategyFactory.deployNewStrategy: Strategy already exists for token");
+        IStrategy strategy = IStrategy(address(
+            new BeaconProxy(
+                address(strategyBeacon),
+                abi.encodeWithSelector(StrategyBase.initialize.selector, token, pauserRegistry)
+            )
+        ));
+        _setStrategyForToken(token, strategy);
+        IStrategy[] memory strategiesToWhitelist = new IStrategy[](1);
+        bool[] memory thirdPartyTransfersForbiddenValues = new bool[](1);
+        strategiesToWhitelist[0] = strategy;
+        thirdPartyTransfersForbiddenValues[0] = false;
+        strategyManager.addStrategiesToDepositWhitelist(strategiesToWhitelist, thirdPartyTransfersForbiddenValues);
```

Comment:
I feel like if the strategyManager takes a dependency on this contract and reads the mapping, we can avoid all of this state copying.

Reply:
it's one-time copying vs. always querying the factory, which amortizes quite quickly to be more efficient.
I don't care too strongly, just felt that not making any changes to the existing core contracts would be simpler to review + faster to execute

Reply:
agreed offline on the approach taken here being OK; this minimizes core contract changes (no changes to StrategyManager), instead opting to turn an existing role (the `strategyWhitelister`) into a smart contract.